File: internal/product/product.go
// Package product holds the product name and version. Rename here and nowhere else.
package product

const (
	Name    = "Infercat" // founder decision 2026-09-05 (docs/PRINCIPLES.md)
	CLIName = "infercat"
	// InvitePrefix is the invite format version tag (docs/ARCHITECTURE.md).
	InvitePrefix = "ic1"
	// WebURL is where a friend opens the web app. Empty until hosting is decided; a host can
	// name their own with `serve --web-url`. When it is known, an invite is printed as a link
	// (<WebURL>#<invite>) that the friend can click instead of copying a code.
	WebURL = "https://infercat.ai"
)

// The build stamp. A release overwrites these at link time (.goreleaser.yaml passes
// -X github.com/infercat/infercat/internal/product.Version=<tag> and friends); a plain
// `go build` keeps the defaults, which is how a developer build says it is one. Version is
// also the single place the Makefile and the web bundle read the version from (docs/PRINCIPLES.md:
// the product name lives in one constant), so keep the literal on one line.
var (
	Version = "0.1.0"
	Commit  = "none"
	Date    = "unknown"
)

File: web/src/product.ts
// The product name lives here and nowhere else on the TS side (docs/PRINCIPLES.md).
// The Go side's copy is internal/product/product.go; keep the two in step by hand.
export const PRODUCT_NAME = 'Infercat';
export const INVITE_PREFIX = 'ic1';
// What the page says about itself to a search engine or a social card (index.html and the web app
// manifest are filled from this line by vite.config.ts). Same words as the connect screen.
export const DESCRIPTION =
  'Chat with a friend’s GPU. They send you one code; you paste it here. No account, no install.';
// Where the code lives: the About line links here. MIT (LICENSE at the repo root).
export const SOURCE_URL = 'https://github.com/infercat/infercat';
// Where a stranger with no code is sent: the README's host quickstart, which is the whole answer to
// "how do I get one of these". The header link and the card's no-code line are the same destination
// and so are one constant, until there is a docs site to point at instead (039).
export const HOST_URL = `${SOURCE_URL}#quickstart-host`;
// The build stamps VITE_APP_VERSION from internal/product/product.go (see the Makefile), so the
// app and the host binary report the same version and a bug report can name both halves. A bare
// `vite build` or `vite dev` has no stamp and says so by falling back to the source default.
export const VERSION: string = import.meta.env.VITE_APP_VERSION ?? '0.0.1-dev';
